golang/src/permissions/restapi/configure_permissions.go

```diff
@@ -0,0 +1,67 @@
+package restapi
+
+import (
+	"crypto/tls"
+	"net/http"
+	"strings"
+
+	errors "github.com/go-swagger/go-swagger/errors"
+	httpkit "github.com/go-swagger/go-swagger/httpkit"
+
+	"permissions/restapi/impl"
+	"permissions/restapi/operations"
+	"permissions/restapi/operations/status"
+)
+
+// This file is safe to edit. Once it exists it will not be overwritten
+
+func configureFlags(api *operations.PermissionsAPI) {
+	// api.CommandLineOptionsGroups = []swag.CommandLineOptionsGroup{ ... }
+}
+
+func configureAPI(api *operations.PermissionsAPI) http.Handler {
+	// configure the api here
+	api.ServeError = errors.ServeError
+
+	api.JSONConsumer = httpkit.JSONConsumer()
+
+	api.JSONProducer = httpkit.JSONProducer()
+
+	api.StatusGetHandler = status.GetHandlerFunc(impl.BuildStatusHandler(SwaggerJSON))
+
+	api.ServerShutdown = func() {}
+
+	return setupGlobalMiddleware(api.Serve(setupMiddlewares))
+}
+
+// The TLS configuration before HTTPS server starts.
+func configureTLS(tlsConfig *tls.Config) {
+	// Make all necessary changes to the TLS configuration here.
+}
+
+// The middleware configuration is for the handler executors. These do not apply to the swagger.json document.
+// The middleware executes after routing but before authentication, binding and validation
+func setupMiddlewares(handler http.Handler) http.Handler {
+	return handler
+}
+
+// The middleware configuration happens before anything, this middleware also applies to serving the swagger.json
+// document. So this is a good place to plug in a panic handling middleware, logging and metrics
+func setupGlobalMiddleware(handler http.Handler) http.Handler {
+	return uiMiddleware(handler)
```

Comment:
Looking at these two functions, it looks initially like the `uiMiddleware` more obviously belongs in the other one (since it doesn't need to happen for serving `swagger.json` itself). I'm guessing this isn't the case because this set of middlewares happens before routing, where the other one happens after, and  we don't want to put `/docs` into the routing stuff since it's not really part of the API. Is this correct?